datacube/virtual/impl.py

```diff
@@ -485,7 +485,12 @@ def group(self, datasets: VirtualDatasetBag, **group_settings: Dict[str, Any]) -
         return self._input.group(datasets, **group_settings)
 
     def fetch(self, grouped: VirtualDatasetBox, **load_settings: Dict[str, Any]) -> xarray.Dataset:
-        return self._transformation.compute(self._input.fetch(grouped, **load_settings))
+        input_data = self._input.fetch(grouped, **load_settings)
+        output_data = self._transformation.compute(input_data)
+        output_data.attrs['crs'] = input_data.attrs['crs']
```

Comment:
I have found setting attributes like this does not always work as it appears to be readonly. As a workaround in `rioxarray` we have an `update_attrs` method. [code link](https://github.com/corteva/rioxarray/blob/4c8844aff5c33656fefd794db8e0cc9d7d0cf8f4/rioxarray/rioxarray.py#L372-L392)

Reply:
It worked on my machine! :tm: 

That method definitely looks cool. Any idea when it does not work?

Reply:
I don't remember exactly when. But, there were instances where I did what you did and the attribute was not added. Does the `crs` attribute already exist when you update it?

Reply:
I don't expect it to in my case. More honestly, I expect it to have the same value if it exists.

Reply:
Alright, in that case, I just recommend keeping an eye out for the issue.